scripts/errata_fix_script.py

```diff
@@ -0,0 +1,199 @@
+                location_href
+                for location_href, pkg in albs_packages_cache.items()
+                if pkg['arch'] in (collection_arch, 'noarch')
+            }
+            errata_packages = {
+                pkg.filename
+                for pkg in collection.packages
+            }
+            missing_filenames = albs_packages.difference(errata_packages)
+            if not missing_filenames:
+                logging.info('Errata %s is ok', record.id)
+                continue
+            for filename in missing_filenames:
+                pulp_pkg = albs_packages_cache[filename]
+                if (
+                    pulp_pkg['sha256'] is None
+                    and pulp_pkg['checksum_type'] == 'sha256'
+                ):
+                    logging.warning(
+                        'sha256 sum is missing for %s using pkgId instead',
+                        filename,
+                    )
+                    pulp_pkg['sha256'] = pulp_pkg['pkgId']
+                collection.packages.append(UpdatePackage(
+                    name=pulp_pkg['name'],
+                    filename=pulp_pkg['location_href'],
+                    arch=pulp_pkg['arch'],
+                    version=pulp_pkg['version'],
+                    release=pulp_pkg['release'],
+                    epoch=str(pulp_pkg['epoch']),
+                    reboot_suggested=pulp_pkg['reboot_suggested'],
+                    src=pulp_pkg['rpm_sourcerpm'],
+                    sum=pulp_pkg['sha256'],
+                    sum_type=cr.checksum_type('sha256'),
+                ))
+                logging.info('Added package %s for errata %s',
+                             filename, record.id)
+            logging.info('Start checking record %s references', record.id)
+            for reference in record.references:
+                if (
+                    reference.title
+                    or reference.ref_type == ErrataReferenceType.bugzilla.value
+                ):
+                    continue
+                reference.title = reference.ref_id
+                logging.info('Fixed ref_title for ref_id %s', reference.ref_id)
+            latest_record_id = record.id
+        pulp_db.commit()
+    logging.info('pulp db records updated')
+    platforms = yaml.safe_load(
+        open("reference_data/platforms.yaml", "r").read()
+    )
+    tasks = []
+    for platform in platforms:
+        for repo in platform["repositories"]:
+            if not repo.get("production"):
+                continue
+            tasks.append(update_pulp_repo(repo, pulp))
+    await asyncio.gather(*tasks)
+    logging.info("Update pulp db is done")
+
+
+async def update_albs_db():
+    logging.info("Update albs db started")
+    query = (
+        select(ErrataReference)
+        .where(
+            ErrataReference.ref_type != ErrataReferenceType.bugzilla,
+            ErrataReference.title == '',
+        )
+    )
+    async for db in get_db():
```

Comment:
you can use `contextlib.asynccontextmanager(get_db)()` construct, this way you'll be able to write whole `with` statement in one string